modules/credential-store/credential-store-service/src/main/java/org/apache/airavata/credential/store/store/impl/util/CredentialStoreInitUtil.java

```diff
@@ -86,6 +98,35 @@ public static void initializeDB() {
                 logger.error("Error while closing database connection...", e.getMessage(), e);
             }
         }
+
+        try {
+
+            GatewayProfileResource gatewayProfileResource = new GatewayProfileResource();
```

Comment:
This should use the Registry API rather than the Resource classes directly.

But I don't think this is the best way to do it. The credential store just stores credentials, it doesn't know anything about the default gateway or the app catalog.  I think the following two places are better:
* https://github.com/apache/airavata/blob/master/modules/registry/registry-server/registry-api-service/src/main/java/org/apache/airavata/registry/api/service/util/AppCatalogInitUtil.java#L101
  * however, this might not work since the credential store is started after the registry, see ServerMain.java
* when the API server starts, [similar to how the sharing registry is initialized](https://github.com/apache/airavata/blob/master/airavata-api/airavata-api-server/src/main/java/org/apache/airavata/api/server/handler/AiravataServerHandler.java#L122)

Reply:
I also initially thought of of putting it to AppCatalogInitUtil but due to the initialization order, it didn't work. I'll move the code to API server which makes more sense.